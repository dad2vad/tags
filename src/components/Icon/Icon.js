import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

class Icon extends PureComponent {
  render() {
    return (
      <span className="icon is-small">
        <i className={`fa fa-${this.props.name}`} />
      </span>
    );
  }
}

Icon.propTypes = {
  name: PropTypes.string.isRequired,
};

export default Icon;
