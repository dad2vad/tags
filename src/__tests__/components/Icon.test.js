import React from 'react';
import { shallow } from 'enzyme';
import Icon from '../../components/Icon';

describe('Icon', () => {
  it('renders correctly', () => {
    const component = shallow(<Icon name="github" />);
    expect(component).toMatchSnapshot();
  });
});
