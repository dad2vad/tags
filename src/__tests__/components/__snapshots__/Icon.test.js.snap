// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders correctly 1`] = `
<span
  className="icon is-small"
>
  <i
    className="fa fa-github"
  />
</span>
`;
